File: GameServer/Models/Bullet.cs
namespace GameServer.Models
{
    public class Bullet
    {
        public Bullet()
        { }

        public Bullet(Bullet otherBullet)
            : this()
        {
            Position = new Vector2(otherBullet.Position);
            Radius = otherBullet.Radius;
            Speed = new Vector2(otherBullet.Speed);
            PlayerId = otherBullet.PlayerId;
            Id = otherBullet.Id;
        }

        public Vector2 Position { get; set; }
        public double Radius { get; set; }
        public Vector2 Speed { get; set; }
        public int PlayerId { get; set; }
        public int Id { get; set; }
    }
}

File: GameServer/wwwroot/js/Physics/Bullet.js
function Bullet(playerId) {

    const that = this;

    that._playerId = playerId;
    that.GetPlayerId = () => that._playerId;

    that._position = new Vector2();
    that.GetPosition = () => that._position;
    that.SetPosition = (value) => that._position = value;

    that.GetX = () => that.GetPosition().GetX();
    that.GetY = () => that.GetPosition().GetY();

    that.SetX = (value) => that._position.SetX(value);
    that.SetY = (value) => that._position.SetY(value);

    that._speed = new Vector2();
    that.GetSpeed = () => that._speed;
    that.SetSpeed = (value) => that._speed = value;

    that._radius = 6;
    that.GetRadius = () => that._radius;
    that.SetRadius = (value) => that._radius = value;

    that._angle = 0;
    that.GetAngle = () => that._angle;
    that.SetAngle = (value) => that._angle = value;
}

// Export module
if (typeof module !== 'undefined' && module.hasOwnProperty('exports')) {
    module.exports = Bullet;
}